pkg/datastore/postgres/repositories/checked_headers_repository.go

```diff
@@ -70,7 +67,7 @@ func (repo CheckedHeadersRepository) UncheckedHeaders(startingBlockNumber, endin
 			           AND block_number >= $1
 			           AND eth_node_id = $3)
 			   OR (check_count < $2
-			           AND last_checked <= ((SELECT MAX(block_number) FROM public.headers) - ($4 * check_count))
+			           AND block_number <= ((SELECT MAX(block_number) FROM public.headers) - ($4 * (check_count * (check_count + 1) / 2)))
```

Comment:
I know this part is a little funky, but it's based on the formula to find the [sum of an arithmetic sequence](https://www.mathplanet.com/education/algebra-2/sequences-and-series/arithmetic-sequences-and-series). E.g. if we want to recheck a header 3 times, then we want the distance between checks to be 15 blocks, then 30 blocks, then 45 blocks.

So to find the _next_ block where a header should be considered "unchecked", we basically need to sum up the sequence `15 + (checkCount - 1) * 15`, or simply `15 * checkCount` for integers from 1 to the current check count, and add the result to the header's block number.

So all this change is really doing is replacing the `last_checked` term with the aforementioned formula for calculating the sum of the arithmetic sequence, and then solving the inequality for `block_number`.

Reply:
Dig it!